Add rudimentary frontend tests for all functionalities

agoscinski opened the pull request:
For the frontend and backend tests new profiles are created to separate the effects of workgraph tests from the GUI.

Also only one workgraph is run for the frontend tests to save time.

Implements request in issue #159

TODO:
- [x] Move settings tests to mutable section (because the settings stop the daemon at the end, they only change the state if unsuccessful so I keep them where they are)

superstar54 reviewed (changes requested):
Better to move the fixtures to the `conftest.py` file. Could you create two folders for backend and frontend, and then create the conftest.py for each.

agoscinski committed:
Add rudimentary frontend tests for all functionalities (#221)

* Implements requested tests in issue #159.

* Backend and frontend tests now use their own dedicated aiida profile to avoid
changes in the aiida database that might affect the tests. For that they have
been moved to their own module folder.

* Frontend tests reuse ran aiida workgraph calculation to save time.

* CI tests have been splitted into three runs of pytest (see issue #225)

superstar54 reviewed (approved):
Nice work!
